ts/src/ellipx.ts

```diff
@@ -0,0 +1,1583 @@
+     */
+    async fetchOrderBook (symbol: string, limit: Int = undefined, params = {}): Promise<OrderBook> {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const marketSymbol = market['symbol'].replace ('/', '_'); // Convert BTC/USDC to BTC_USDC
+        const request = {
+            'currencyPair': marketSymbol,
+        };
+        const response = await this.publicGetMarketCurrencyPairGetDepth (this.extend (request, params));
+        // {
+        //     "data": {
+        //         "asks": [
+        //             {
+        //                 "price": {
+        //                     "v": "74941875231",
+        //                     "e": 6,
+        //                     "f": 74941.875231
+        //                 },
+        //                 "amount": {
+        //                     "v": "149",
+        //                     "e": 8,
+        //                     "f": 0.00000149
+        //                 }
+        //             },
+        //             {
+        //                 "price": {
+        //                     "v": "75063426037",
+        //                     "e": 6,
+        //                     "f": 75063.426037
+        //                 },
+        //                 "amount": {
+        //                     "v": "335",
+        //                     "e": 8,
+        //                     "f": 0.00000335
+        //                 }
+        //             }
+        //         ],
+        //         "bids": [
+        //             {
+        //                 "price": {
+        //                     "v": "64518711040",
+        //                     "e": 6,
+        //                     "f": 64518.71104
+        //                 },
+        //                 "amount": {
+        //                     "v": "132",
+        //                     "e": 8,
+        //                     "f": 0.00000132
+        //                 }
+        //             },
+        //             {
+        //                 "price": {
+        //                     "v": "64263569273",
+        //                     "e": 6,
+        //                     "f": 64263.569273
+        //                 },
+        //                 "amount": {
+        //                     "v": "210",
+        //                     "e": 8,
+        //                     "f": 0.0000021
+        //                 }
+        //             }
+        //         ],
+        //         "market": "BTC_USDC"
+        //     },
+        //     "request_id": "71b7dffc-3120-4e46-a0bb-49ece5aea7e1",
+        //     "result": "success",
+        //     "time": 0.000074661
+        // }
+        if (response['result'] !== 'success') {
+            throw new ExchangeError (this.id + ' fetchOrderBook() failed: ' + this.json (response));
```

Comment:
again all error handling should be done inside handleErrors and not here

Reply:
Got it, will remove all error handling inside each function and implement `handleErrors` as working through these function.